### ISSUE TITLE
Use system DNS server instead of proxy tunnel

### ISSUE THREAD

**Issue by HikariWS** [FelisCatus/SwitchyOmega#2115]
### SwitchyOmega version / SwitchyOmega 版本
2.5.21

### Browser version & OS version / 浏览器名称、版本及操作系统版本
Vivaldi 3.4.2066.94 (Stable channel) (64-bit)
Windows 10 OS Version 1903 (Build 18362.836)

### Problem description / 问题描述
I have Pihole on a local server. When I'm using a HTTP proxy profile, my DNS server is ignored and another one, IDK which, is used.

(Please provide as much detail as possible. We recommend the following format.)
(请尽可能多提供一些细节。我们推荐使用下面的格式。)

#### Steps to reproduce issue / 重现错误所需步骤

(What did you do? / 你做了什么？)

1. Set System Proxy and enter a domain blocked on Pihole
2. Browser says "server IP address could not be found"
3. Set any HTTP proxy or auto switch
4. The site opens normally

#### Expected behavior / 期望发生的情况
Browser should say "server IP address could not be found" even when a proxy is used.

#### Actual (or suggested) behavior / 实际发生的情况（或建议修改后的行为）
I'd like to be able to set on each profile if it should use system DNS or tunnel DNS queries on the proxy.

I also have a Tor proxy, and for it the .onion query must be tunneled so hidden services are resolved and reachable. Because of that, it's needed to set the DNS behavior by profile.

**Comment by yfdyh000** [comment]
This is impossible, the HTTP proxy always uses the remote parser.

**Comment by HikariWS** [comment]
It may not be available on a software, but cannot be impossible. Specially when the software is able to choose from a variety of proxies which one to use.

**Comment by AoiKagase** [comment]
I think this feature is necessary if you take advantage of SOCKS5.
The FoxyProxy add-on for Firefox allows you to check "Send DNS through SOCKS5 proxy" when you select SOCKS5.
By unchecking it, you can use your local DNS.

**Comment by HikariWS** [comment]
Yeah, that would be the best solution, to let us choose for each profile if DNS should be tunneled to proxy server or use system settings.

**Comment by HikariWS** [comment]
I think that even better than just let us choose to use local settings or external settings, we could just set the DNS server IP address for each profile, and if to reach it from inside or outisde proxy tunnel.

In this case IDK if proxy software is able to do that, but from UX perspective it'd be great.

I'm thinking now and there are use cases we do need to use proxy even when it's HTTPS. For some time on my office work, some development sites were blocked because company understood they were blogs and I'd not be working. If some sites are blocked on network local DNS, we do need to use proxy to query DNS too. I guess it uses proxy server's settings, it'd be better if we could choose which DNS server to use after leaving the tunnel.

**Comment by IntoTheBlackHole** [comment]
A temporary workaround I am using is FoxyProxy for firefox
